src/main/java/com/cloudbees/jenkins/support/filter/ContentMappings.java

```diff
@@ -141,9 +141,12 @@ private ContentMappings(@Nonnull XmlProxy proxy) {
      * Looks up or creates a new ContentMapping for the given original string and a ContentMapping generator.
      */
     public @Nonnull ContentMapping getMappingOrCreate(@Nonnull String original, @Nonnull Function<String, ContentMapping> generator) {
+        boolean isNew = !mappings.containsKey(original);
         ContentMapping mapping = mappings.computeIfAbsent(original, generator);
```

Comment:
Maybe you could use a new generator to avoid having to inspect for the keys. If I am not wrong, the generator is called only if necessary, and you could create onw that would: set isNew to true and then call the real generator.

Reply:
It's not possible because the _boolean_ should be final and its value could not be updated